plaidml/edsl/ffi.cc

```diff
@@ -973,7 +976,16 @@ plaidml_program* plaidml_program_evaluate(  //
       }
       mutations.updates.emplace(ProgramUpdate{src_updates[i]->value, dst_updates[i]->value});
     }
-    auto ret = new plaidml_program{GlobalContext::get()->MakeProgram(name, mutations)};
+
+    if (!plaidml::isFloat(static_cast<plaidml::DType>(floatx))) {
```

Comment:
When we're on the `ffi` side we shouldn't be using the `plaidml` API (that's only for the user's side). Once we get across the C boundary, we should use internal types, so let's cast the `plaidml_datatype` directly into a `pmlc::util::DataType` (the enum values are intentionally identical between the 2 definitions to allow this cheap/easy conversion).